controller/api/public/grpc_server.go

```diff
@@ -120,13 +121,26 @@ func (s *grpcServer) ListPods(ctx context.Context, req *pb.ListPodsRequest) (*pb
 		controllerComponent := pod.Labels[pkgK8s.ControllerComponentLabel]
 		controllerNS := pod.Labels[pkgK8s.ControllerNSLabel]
 
+		proxyReady := false
+		proxyVersion := ""
+		for _, container := range pod.Status.ContainerStatuses {
+			if container.Name == pkgK8s.ProxyContainerName {
+				proxyReady = container.Ready
+
+				parts := strings.Split(container.Image, ":")
```

Comment:
TIL that the `pod.status.containerStatuses[].image` field in the pod API response isn't guaranteed to match the `pod.spec.containers[].image` field. In my environment I have the same proxy image tagged multiple times, and as far as I can tell, the status container arbitrarily picks which image tag it uses, whereas the spec container reflects what's actually in the spec. So for instance, when I look at my injected controller pod, I see different strings for the proxy's spec container and the status container:

```
$ curl -s localhost:8001/api/v1/namespaces/linkerd/pods/controller-7b9df96664-lxm9p | jq -S . | grep 'proxy:'
        "image": "gcr.io/linkerd-io/proxy:git-05e6d58e",
        "image": "gcr.io/linkerd-io/proxy:dev-37884805-kl",
```

So I think we need to use the spec container when populating the image field in the pod struct. This should do it:

```
		proxyReady := false
		for _, container := range pod.Status.ContainerStatuses {
			if container.Name == pkgK8s.ProxyContainerName {
				proxyReady = container.Ready
			}
		}

		proxyVersion := ""
		for _, container := range pod.Spec.Containers {
			if container.Name == pkgK8s.ProxyContainerName {
				parts := strings.Split(container.Image, ":")
				proxyVersion = parts[1]
			}
		}
```

Reply:
good catch! fixed